doc/src/guides/contributing/dependencies.md

```diff
@@ -0,0 +1,222 @@
+[conda-forge](https://conda-forge.org/), and most can also be installed with
+`pip`.
+
+This page does not list packages which themselves depend on SymPy, only those
+packages that SymPy depends on. An incomplete list of packages that depend on
+SymPy can be found on the [main SymPy
+webpage](https://www.sympy.org/en/index.html), and a more complete list can be
+found on
+[GitHub](https://github.com/sympy/sympy/network/dependents?dependent_type=PACKAGE)
+or [libraries.io](https://libraries.io/pypi/sympy/dependents).
+
+## Hard Dependencies
+
+SymPy only has one hard dependency, which is required for it to work: mpmath.
+
+### mpmath
+
+[mpmath](https://mpmath.org/) is a pure Python package for arbitrary precision
+arithmetic. It is used under the hood whenever SymPy calculates the
+floating-point value of a function, e.g., when using
+[evalf](sympy.core.evalf.EvalfMixin.evalf).
+
+SymPy cannot function with mpmath and will fail to import if it is not
+installed. If you get an error like
+
+```
+ImportError: SymPy now depends on mpmath as an external library. See
+https://docs.sympy.org/latest/install.html#mpmath for more information.
+```
+
+this means that you did not install mpmath correctly. [This
+page](mpmath-install) explains how to install it.
+
+Most methods of installing SymPy, such as the ones outlined in the
+[installation](installation) guide, will install mpmath automatically. You
+typically only need to install mpmath manually if you did not actually install
+SymPy, e.g., if you are developing directly on SymPy in the git repository.
+
+## Optional Dependencies
+
+These dependencies are not required to use SymPy. The vast majority of SymPy
+functions do not require them, however, a few functions such as plotting and
+automatic wrapping of code generated functions require additional dependencies
+to function.
+
+Additionally, as a contributor, when running the SymPy tests, some tests will
+be skipped if a dependency they require is not installed. The [GitHub Actions
+CI](https://github.com/sympy/sympy/actions) which is run on every SymPy pull
+request will automatically install these dependencies in the
+"optional-dependencies" build, but you may wish to install them locally if you
+are working on a part of SymPy that uses them.
+
+### Recommended Optional Dependencies
+
+These dependencies are not required for SymPy to function, but it is
+recommended that all users install them if they can, as they will improve the
+general performance of SymPy.
+
+#### gmpy2
+
+[gmpy2](https://gmpy2.readthedocs.io/en/latest/intro.html) is a Python wrapper
+for the GMP multiple-precision library. It provides large integers that are
+faster than the built-in Python `int`. When gmpy2 is installed, it is used
+automatically by certain core functions that operate on integers, such as the
+[polys](polys-docs). See {ref}`polys-domainsref` for more details.
+
+### Interactive Use
+
+#### IPython
+
+#### Jupyter Notebook
+
```

Comment:
Mention Spyder?

Reply:
There's no spyder specific code in SymPy itself (that I'm aware of).

Reply:
Is there any Jupyter Notebook-specific?

Reply:
Yes, we have hooks for MathJax printing in the Jupyter notebook (enabled by default, and customizable with `init_printing()`).